commit 9f48ed2e509ee80d39e3018cadf4b2c6a4a28ee9 (refs/changes/60/1260/2)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2017-01-17T16:19:16-08:00 (2 years, 9 months ago)
    
    DOCKER-944 Docker pull fails with vague error message with insecure registries

diff --git a/lib/docker-stdio.js b/lib/docker-stdio.js
index 784116b..9a2029b 100644
--- a/lib/docker-stdio.js
+++ b/lib/docker-stdio.js
@@ -25,7 +25,6 @@ var http = require('http');
 var net = require('net');
 var path = require('path');
 var pty = require('pty.js');
-var shellescape = require('shell-escape');
 var spawn = child_process.spawn;
 var execFile = child_process.execFile;
 var vmadm = require('vmadm');
@@ -300,13 +299,16 @@ function createDockerFileReadStreamServer(opts, callback) {
         var tar = ['/native/usr/bin/gtar', 'cf', '-'];
 
         tar.push('-C', path.dirname(norm), path.basename(norm));
-        tar = arrayFromShellEscape(tar);
 
         var zloginTarCmd = '/usr/sbin/zlogin';
         var zloginTarArgs = ['-Q', uuid];
 
         Array.prototype.push.apply(zloginTarArgs, tar);
 
+        log.info({ zloginTarArgs: zloginTarArgs },
+                 'createDockerFileReadStreamServer ' +
+                 'onConnectionZoneRunning zloginTarArgs');
+
         var streamProc =
             spawn(zloginTarCmd, zloginTarArgs, { encoding: 'binary' });
 
@@ -495,7 +497,6 @@ function createDockerFileWriteStreamServer(opts, callback) {
             '--no-overwrite-dir' : '--overwrite';
 
         var tar = ['/native/usr/bin/gtar', 'xf', '-', overwrite, '-C', norm];
-        tar = arrayFromShellEscape(tar);
 
         var zloginTarCmd = '/usr/sbin/zlogin';
         var zloginTarArgs = ['-Q', uuid];
@@ -1203,22 +1204,6 @@ function dockerPathStatFromPath(opts) {
 }
 
 
-function arrayFromShellEscape(a) {
-    var ret = [];
-
-    a.forEach(function (s) {
-        if ((new RegExp('[^A-Za-z0-9_/:=-]')).test(s)) {
-            s = '\'' + s.replace(new RegExp('\'', 'g'), '\'\\\'\'') + '\'';
-            s = s.replace(new RegExp('^(?:\'\')+', 'g'), '')
-                 .replace(new RegExp('\\\'\'\'', 'g'), '\\\'');
-        }
-        ret.push(s);
-    });
-
-    return ret;
-}
-
-
 module.exports = {
     setupDockerExecution: setupDockerExecution,
     setupDockerFileStream: setupDockerFileStream
diff --git a/package.json b/package.json
index a762eab..3cb5a74 100644
--- a/package.json
+++ b/package.json
@@ -26,7 +26,6 @@
     "sdc-clients": "10.0.3",
     "sdc-docker-build": "git+https://github.com/joyent/sdc-docker-build.git#af8bd3e",
     "semver": "2.2.1",
-    "shell-escape": "^0.2.0",
     "sprintf": "0.1.5",
     "trace-event": "git+https://github.com/joyent/node-trace-event.git#9f7d00b8c3594def9ef534f68c16c215c3dba9f2",
     "tracker": "git+https://github.com/joyent/node-tracker.git#3a139906c9eb8d8684ac54cf54cd010315856042",
